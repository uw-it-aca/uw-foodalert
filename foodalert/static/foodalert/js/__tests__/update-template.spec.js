"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount} from '@vue/test-utils';
import UpdateController from '../components/update-controller.vue';
const $route = {
    query: {
        parent_notification: 1
    }
}
const $v = {
    form: {
        text: {
            required: true,
            maxLength: true
        },
        $error: false
    }
}

describe('Update Controller', function () {
    describe('renders correctly', function () {
        test('textarea', function () {
            var wrapper = mount(UpdateController, {
                mocks: {
                    $route,
                    $v
                },
            });
            expect(wrapper.element).toMatchSnapshot();
        });
    });
})
