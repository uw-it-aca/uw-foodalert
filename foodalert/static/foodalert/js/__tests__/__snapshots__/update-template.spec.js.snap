// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Controller renders correctly 1`] = `
<div
  class="px-4 pt-3 container"
  text=""
>
  <p>
    <strong>
      Thanks for Sharing! Here's what's next:
    </strong>
  </p>
   
  <!---->
   
  <p
    style="color: red;"
  >
    You must be currently hosting an event to submit an update
  </p>
   
  <p
    class="mt-4"
  >
     Let people know when food runs out 
  </p>
   
  <button
    class="btn w-100 mb-3 py-2 btn-primary disabled"
    disabled="disabled"
    type="button"
  >
     No Food Left 
  </button>
   
  <div
    class="px-0 py-1 mt-3 container-fluid"
    is-optional=""
  >
    <div
      aria-labelledby="__BVID__4__BV_label_"
      class="labelled-textarea mb-0 b-form-group form-group"
      id="__BVID__4"
      role="group"
    >
      <label
        class="col-form-label pt-0"
        for="textarea-null"
        id="__BVID__4__BV_label_"
      >
        
            Notify people if plans change 
        <br />
      </label>
      <div
        class=""
      >
         
        <textarea
          class="form-control form-control-md"
          id="textarea-null"
          placeholder="e.g. We moved to HUB 230! Still 4 more sandwiches"
          rows="2"
          wrap="soft"
        />
        <!---->
        <!---->
        <!---->
      </div>
    </div>
     
    <!---->
  </div>
   
  <p>
     
  </p>
   
  <!---->
   
  <a
    aria-disabled="true"
    class="w-100 mb-3 py-2 btn btn-secondary btn-md disabled"
    href="#"
    tabindex="-1"
    target="_self"
    type="submit"
  >
     Send Update 
  </a>
   
  <div
    class="text-center"
  >
    <div
      id="__BVID__6___BV_modal_outer_"
    >
      <div
        aria-hidden="true"
        class="modal fade"
        id="__BVID__6"
        role="dialog"
        style="display: none;"
      >
        <div
          class="modal-dialog modal-md"
        >
          <div
            aria-describedby="__BVID__6___BV_modal_body_"
            aria-labelledby="__BVID__6___BV_modal_header_"
            class="modal-content"
            role="document"
            tabindex="-1"
          >
            <header
              class="modal-header border-0"
              id="__BVID__6___BV_modal_header_"
            >
              <h5
                class="modal-title"
              />
              <button
                aria-label="Close"
                class="close"
                type="button"
              >
                ×
              </button>
            </header>
            <div
              class="modal-body"
              id="__BVID__6___BV_modal_body_"
            >
              <p
                class="h5 pb-3"
              >
                 Are you sure you'd like to notify student's there's 
                <strong>
                   no food left
                </strong>
                ? 
              </p>
               
              <blockquote
                class="border rounded mb-0 mx-3 p-3 text-left bg-light"
              >
                Update: No Food left! The event: “” has ended and is no longer serving food.
              </blockquote>
               
            </div>
            <footer
              class="modal-footer border-0 d-flex flex-column"
              id="__BVID__6___BV_modal_footer_"
            >
              <a
                aria-disabled="true"
                class="btn btn-lg btn-primary mx-auto mb-2 disabled"
                href="#"
                tabindex="-1"
                target="_self"
              >
                 Send 'No Food Left' 
              </a>
               
              <a
                class="mx-auto"
                href="#"
              >
                 Cancel 
              </a>
            </footer>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
